cli/azd/internal/repository/app_init.go

```diff
@@ -359,16 +375,20 @@ func (i *Initializer) genProjectFile(
 
 const InitGenTemplateId = "azd-init"
 
-func prjConfigFromDetect(
+func (i *Initializer) prjConfigFromDetect(
```

Comment:
nit. I'm not a fan of the super long methods.  Is this almost 200?

Reply:
Let's also think about the cost of separation tradeoff here: would a bug be as easily caught in review when the function is broken into 5 separate functions, each with their own input/output contracts, and data is flowing through them?